networkx/drawing/nx_pydot.py

```diff
@@ -182,8 +168,10 @@ def to_pydot(N, strict=True):
     -----
 
     """
-    pydot = load_pydot()
-
+    try:
+        import pydotplus as pydot
+    except ImportError:
+        raise ImportError("pydotplus required for to_pydot()")
```

Comment:
`nx.drawing` does a full import of this module. A top-level import will cause an ImportError even though pydot is considered an optional dependency. We *could* catch the error in `networkx.drawing.__init__.py` and the functions would simply be absent.  I think we avoided that in the past since it caused confusion as to whether the function was missing due to having the wrong NX version or due to a missing optional dependency.  The function-level import allows us to present the same API whether or not the optional dependency is present. And in the (probably) rare situation where someone imports networkx without pydot, sees the import error, backgrounds the process to pip install pydot, and tries the pydot function again, then it will just work.